src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2376,555 +2376,723 @@ protected virtual void MarkReflectionLikeDependencies (MethodBody body)
 				if (methodCalledDefinition == null)
 					continue;
 
-				var methodCalledType = methodCalled.DeclaringType;
-
-				int first_arg_instr, second_arg_instr;
-				Instruction first_arg;
-
-				switch (methodCalledType.Name) {
-				//
-				// System.Type
-				//
-				case "Type" when methodCalledType.Namespace == "System":
-
-					// Some of the overloads are implemented by calling another overload of the same name.
-					// These "internal" calls are not interesting to analyze, the outermost call is the one
-					// which needs to be analyzed. The assumption is that all overloads have the same semantics.
-					// (for example that all overload of GetConstructor if used require the specified type to have a .ctor).
-					if (methodCalling.DeclaringType == methodCalling.DeclaringType && methodCalling.Name == methodCalled.Name)
-						break;
+				ReflectionPatternContext reflectionContext = new ReflectionPatternContext (_context, body.Method, methodCalledDefinition);
```

Comment:
Because for some reason `using` doesn't work on structs... compiler simply says that it won't allow it. I figured that since it's only in this one place, it's OKish to do the try/finally instead.

Reply:
Sorry - that was incorrect - using(Struct) works, but it makes the struct readonly. That means I can't pass it as a ref parameter (since the callee could mutate it). I need the ref parameter to avoid copying (since I'm using the struct as a way to allocate on stack, but I still need a single instance only), so can't use `using`.

https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1657